lua/apple-music/init.lua

```diff
@@ -242,6 +256,18 @@ M.unfavorite_current_track = function()
 	set_track_favorited_state(current_track, false)
 end
 
+---Toggle favorite for current (playing) track.
+---NOTE: Below macOS 14 (Sonoma), it'll toggle `loved`.
+---@usage require('apple-music').toggle_favorite_current_track()
+M.toggle_favorite_current_track = function()
```

Comment:
```suggestion
---@usage require('apple-music').toggle_favorite()
M.toggle_favorite = function()
```

Reply:
I don't know if `current_track` is obvious or not. If you don't think so don't apply this.

Reply:
In that case, I'd rather transform this function in a way so we can give it a track name (as the local functions already support that). So if we have a parameter (`track`) that is not nil then toggle that one else do it for the current track. (also change the other two fav functions like this)

Then we could use it like this:
`vim.keymap.set("n", "<leader>amt", apple_music.toggle_favorite)`

Or, use it somewhere else by providing a track name (e.g. I could imagine it in telescope)

What do you think of it?

Reply:
You're probably right about that if we changed it to just `toggle_favorite()`. That being said, I think I prefer the way you did it originally here instead of having optional parameters that change the behavior.

Feel free to merge as is. 👍

Reply:
Okay, I think I'll create a new GH issue to make the above-mentioned change later. Thanks! 😎

Reply:
Btw, seems like only you can merge it 

![Screenshot 2024-07-29 at 22 34 24](https://github.com/user-attachments/assets/490effbe-e78e-47b1-9cb1-96994e5e2d80)

Reply:
I stand corrected... 🫡 


> Okay, I think I'll create a new GH issue to make the above-mentioned change later.

I actually like the way you did it. I don't think we need to change it. If there's ever a 2.0 and we consider redesigning the entire API we can think about it 😉 .